scripts/nix-profile.sh.in

```diff
@@ -51,9 +55,24 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     # pick up `.nix-profile/share/man` because is it close to `.nix-profile/bin`
     # which is in the $PATH. For more info, run `manpath -d`.
     if [ -n "${MANPATH-}" ]; then
-        export MANPATH="$NIX_LINK/share/man:$MANPATH"
+        # only append once
+        case ":$MANPATH:" in
+          *:"$NIX_LINK/share/man":*)
+            ;;
+          *)
+            export MANPATH="$NIX_LINK/share/man:$MANPATH"
+            ;;
+        esac
     fi
```

Comment:
I'm not sure how well this will work. When I run `echo $MANPATH`, I see this in kitty:

```
/Applications/kitty.app/Contents/Resources/man:
```

And get empty output in Terminal.

This is in a session where both the default and my user profile have been sourced and man can access manpages installed by nix.

Running `manpath -d`:

```
-- Searching PATH for man directories
--   Adding /Users/feuh/.nix-profile/share/man to manpath
--   Adding /nix/var/nix/profiles/default/share/man to manpath
--   Adding /usr/share/man to manpath
--   Adding /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/share/man to manpath
--   Adding /Library/Developer/CommandLineTools/usr/share/man to manpath
-- Adding default manpath entries
-- Parsing config file: /etc/man.conf
-- Using manual path: /Users/feuh/.nix-profile/share/man:/nix/var/nix/profiles/default/share/man:/usr/share/man:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/share/man:/Library/Developer/CommandLineTools/usr/share/man
/Users/feuh/.nix-profile/share/man:/nix/var/nix/profiles/default/share/man:/usr/share/man:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/share/man:/Library/Developer/CommandLineTools/usr/share/man
```

But the environment variable is still empty. Very strange.

Reply:
I think what you're observing here is that the block will only operate if MANPATH is already set when the shell hook runs, because it's guarded with:

```bash
if [ -n "${MANPATH-}" ]; then
```

This doesn't sound like a problem, though I'm less certain about whether a knock-on implication of your observation  and this change would be a problem: 

Moving Nix's shell hook from /etc/zshrc to /etc/zshenv does move it up in the execution order, which in turn comes with some risk of shifting behavior for any users that explicitly set MANPATH before the `/etc/zshrc` hook runs, but not before the `/etc/zshenv` hook would run.

Reply:
Hmm that seems to only be a problem if the order of paths in PATH and MANPATH differs. In that case, you could hit a situation where by order of PATH, you execute one version of a program, but by order of MANPATH, the manpage of a different version would be shown. On macOS in particular, the highest disruption around this would probably be related to coreutils, where the BSD versions are used instead of the GNU versions, and git, where Apple provides their own fork that interacts well with the macOS keychain. In these cases, though, they're installed system-wide, and their PATH is not mentioned in `/etc/zshrc`.

However, if PATH (or MANPATH) were to be set before the nix hook, that could indeed change the behavior.

So the only situation where this change would cause an effect is if a user changed `/etc/zshrc` to contain something like this:

```
PATH=/home/myself/repos/myprogram/bin:$PATH

# Nix
if [ -e /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh ]; then
  . /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
# End Nix
```

This seems exceedingly unlikely though. Normally you customize the `zshrc` of your user, and if you want one of your paths to take precedence, you just add it in the front:

```
PATH=$PATH:/home/myself/repos/myprogram/bin
```

I would say the positive impact of this change far outweighs the small likelyhood that this might potentially break some obscure configuration change that the next OS update would have wiped out anyway.